src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -282,16 +281,27 @@ public bool TryGetNullableDisabledSemanticModel([NotNullWhen(returnValue: true)]
         /// cref="SupportsSemanticModel"/> returns <see langword="false"/>. This function will
         /// return the same value if called multiple times.
         /// </returns>
-        [Experimental(RoslynExperiments.NullableDisabledSemanticModel, UrlFormat = RoslynExperiments.NullableDisabledSemanticModel_Url)]
-#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
-        public async Task<SemanticModel?> GetSemanticModelAsync(SemanticModelOptions options, CancellationToken cancellationToken = default)
-#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
+#pragma warning disable RSEXPERIMENTAL001 // sym-shipped usage of experimental API
+        internal async Task<SemanticModel?> GetSemanticModelAsync(SemanticModelOptions options, CancellationToken cancellationToken = default)
         {
             return await GetSemanticModelHelperAsync(
                 disableNullableAnalysis: (options & SemanticModelOptions.DisableNullableAnalysis) == SemanticModelOptions.DisableNullableAnalysis,
-                cancellationToken: cancellationToken
+                cancellationToken
             ).ConfigureAwait(false);
         }
+#pragma warning restore RSEXPERIMENTAL001
+
+        // If the API used in this method is made public, we can consider moving this method to DocumentExtensions.
+        internal async ValueTask<SemanticModel> GetRequiredNullableDisabledSemanticModelAsync(CancellationToken cancellationToken)
+        {
+            if (this.TryGetNullableDisabledSemanticModel(out var semanticModel))
+                return semanticModel;
+
+#pragma warning disable RSEXPERIMENTAL001 // Sym-shipped usage of experimental API
```

Comment:
remove these?

Reply:
We need it in order to use the SemanticModelOptions type.

Reply:
wfm.